#! /bin/sh /usr/share/dpatch/dpatch-run
## testsuite-exit-code.dpatch by  <mpitt@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: If any test fails, exit with nonzero. This makes it easier to integrate into package build scripts.
## DP: Forwarded to http://www.cups.org/str.php?L2758

@DPATCH@
diff -urNad trunk~/test/run-stp-tests.sh trunk/test/run-stp-tests.sh
--- trunk~/test/run-stp-tests.sh	2008-03-16 19:55:04.000000000 +0100
+++ trunk/test/run-stp-tests.sh	2008-03-16 19:57:39.000000000 +0100
@@ -590,14 +590,18 @@
 
 if test $fail != 0; then
 	echo "$fail tests failed."
+        EXITCODE=1
 else
 	echo "All tests were successful."
+        EXITCODE=0
 fi
 
 echo "Log files can be found in /tmp/cups-$user/log."
 echo "A HTML report was created in test/$strfile."
 echo ""
 
+exit $EXITCODE
+
 #
 # End of "$Id: run-stp-tests.sh 6649 2007-07-11 21:46:42Z mike $"
 #
